provider/cluster/manager.go

```diff
@@ -194,11 +195,13 @@ func (dm *deploymentManager) startTeardown() <-chan error {
 }
 
 func (dm *deploymentManager) doDeploy() error {
-	return dm.client.Deploy(dm.lease, dm.mgroup)
+	ctx := context.Background() // TODO: manage contexted within the  deploymentManager
```

Comment:
same as before - in `run()`:

```go
ctx, cancel := context.WithCancel(context.Background())
// ...
loop:
for {
// ...
}
cancel()
// ...

Reply:
I started going down that path, between the anonymous functions being initialized in goroutines which would need to take the context as a parameter, and then the builder functions which would also need to take context as a parameter, the state managing functions would need to take context as a parameter... all of this makes for tightly coupled and brittle code, which is not what this issue was about. I think we should punt on updating `deploymentManager` in this changeset.

Reply:
agreed - was a suggestion for next round.

Reply:
fwiw it'd look something like [this](https://github.com/ovrclk/akash/compare/cluster-manager-ctx-example#diff-eebca299d9c7076c0017fb68023f0ca4)

Reply:
also agree that there's probably better ways to structure that manager.